### ISSUE TITLE
 how to use TripletLoss_WRT

### ISSUE THREAD

**Issue by szxczyc** [KaiyangZhou/deep-person-reid#358]
Hi ,kaiyang:
i want to reproduce the AGW method and i found Weighted Regularization Triplet return loss and correct ,so should i rewrite a new image engine follow the torchreid [guide?](https://kaiyangzhou.github.io/deep-person-reid/user_guide#use-torchreid-as-a-feature-extractor-in-your-projects)
```
class TripletLoss_WRT(nn.Module):
    """Weighted Regularized Triplet'."""

        ......

        dist_ap = dist_mat * is_pos
        dist_an = dist_mat * is_neg

        weights_ap = softmax_weights(dist_ap, is_pos)
        weights_an = softmax_weights(-dist_an, is_neg)
        furthest_positive = torch.sum(dist_ap * weights_ap, dim=1)
        closest_negative = torch.sum(dist_an * weights_an, dim=1)

        y = furthest_positive.new().resize_as_(furthest_positive).fill_(1)
        loss = self.ranking_loss(closest_negative - furthest_positive, y)


        # compute accuracy
        correct = torch.ge(closest_negative, furthest_positive).sum().item()
        return loss, correct
```

**Comment by KaiyangZhou** [comment]
you can follow the existing engine code, e.g. [this](https://github.com/KaiyangZhou/deep-person-reid/blob/master/torchreid/engine/image/triplet.py), to write your own engine, simply to inherit the `Engine` class so you don't need to rewrite everything
